Adds custom energy bin support for Fermi/GBM TimeSeries (#7943)
sunpy/sunpy: changelog/7943.feature.rst

> Added a function ``sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries.energy_bins`` to give `sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries` with custom energy bins.

Reviewer: ```suggestion
Added a function ``sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries.energy_bins`` to create a new `sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries` with user-specified energy bins.
```

> Added a keyword ``energy_bands`` for creating `sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries` with custom energy bins while initializing it.

Reviewer: ```suggestion
You can now pass a keyword ``energy_bands`` to create a `sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries` with custom energy bins.
```

> Added a keyword `energy_bands` for creating `sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries` with custom energy bins while initializing it.

Reviewer: Doc build is failing on:

```
<unknown>:7: WARNING: py:func reference target not found: sunpy.timeseries.sources.fermi_gbm.GBMSummaryTimeSeries.energy_bins [ref.func]
<unknown>:7: WARNING: py:obj reference target not found: energy_bands [ref.obj]
```

Author: i think using the same approach as [#6437](https://github.com/sunpy/sunpy/pull/6437/commits/eeb364c22709cbb26ea715141660eb1650a3634e) will fix the docs issue
